### PR TITLE
[SILGen] Cope with protocol requirement overrides for keypaths.

### PR DESCRIPTION
**Pull request by DougGregor**
When emitting accessor calls for keypaths, make sure that we reference
protocol requirements that introduce witness table entries. Other
protocol requirements don't have the necessary dispatch thunks,
resulting in linker errors.

Fixes rdar://problem/44187969.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test macOS

File: test/SILGen/keypath_witness_overrides.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+// CHECK-LABEL: sil hidden @$S25keypath_witness_overrides18getWritableKeyPath_5indexs03AnyfG0Cx_5IndexQzt09protocol_C015OverridesSetterRzSHAGRQlF
+func getWritableKeyPath<OS: OverridesSetter>(_ c: OS, index: OS.Index) -> AnyKeyPath
+where OS.Index: Hashable {
+  // CHECK: keypath $WritableKeyPath<OS, OS.Element>,
+  // CHECK-SAME: id #OverridesSetter.subscript!getter.1
+  // CHECK-SAME: getter @$S18protocol_overrides14OriginalGetterPy7ElementQz5IndexQzcipAA15OverridesSetterRzSHAGRQlxxTK
+  // CHECK-SAME: setter @$S18protocol_overrides10AddsSetterPy7ElementQz5IndexQzcipAA09OverridesD0RzSHAGRQlxxTk
+  // CHECK-SAME: indices_equals @$S5Index18protocol_overrides14OriginalGetterPQzAB15OverridesSetterRzSHAERQlTH
+  // CHECK-SAMEL :indices_hash @$S5Index18protocol_overrides14OriginalGetterPQzAB15OverridesSetterRzSHAERQlTh
```
Comment (jckarter) [inline test/SILGen/keypath_witness_overrides.swift]: Looks like this CHECK-SAME line got garbled.

**Review by jckarter — APPROVED** [review]
Looks good, thanks!

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Comment by DougGregor** [comment]
@swift-ci please smoke test and merge

**Pull request merged**